cnd/src/network/comit_ln.rs

```diff
@@ -306,18 +306,47 @@ impl NetworkBehaviourEventProcess<announce::behaviour::BehaviourOutEvent> for Co
     fn inject_event(&mut self, event: announce::behaviour::BehaviourOutEvent) {
         match event {
             announce::behaviour::BehaviourOutEvent::ReceivedAnnouncement { peer, mut io } => {
+                if let Some(local_swap_id) =
```

Comment:
Note that initially I munched the error inside the 
```
if let Some(local_swap_id) =
                    self.swaps_waiting_for_announcement.remove(&io.swap_digest)	                    self.swaps_waiting_for_announcement.remove(&io.swap_digest)
```

block. 

But that has the disadvantage that we remove the event for a few milliseconds (and then add it again).

I feel the current solution is the cleanest, as it first checks if there are any error without modifying the HashMap and only starts modifying if there are no errors.

Reply:
Can we have some comments in the code please about this.  I was stumped as to why the error handling was all at the top of the function until I read this comment here, it did not occur to me that we would have to add the swap back into the hash map while reading the code.

nit: We don't need the second `if let Some(local_swap_id)` construct, we could use `local_swap_id = self.swaps_waiting_for_announcement.remove(&io.swap_digest).unwrap()` with a comment saying `// unwrap is ok since we check this above` or some such thing.

Reply:
I added comments.
Additionally changed the logic to `match` instead of `if let` because I think it makes it more readable. (then the `else` case is clearly `None`)